### ISSUE TITLE
[BUG] Unable to turn on/off Shelly 2.5 - Discovery works

### ISSUE THREAD

**Issue by homietw** [StyraHem/ShellyForHASS#319]
## Environment

- ShellyForHASS release with the issue: 0.1.9 Beta 2 and 0.1.8
- Last working ShellyForHASS release (if known): 
- Home Assistant Core release with the issue: 0.110.5
- Operating environment (Home Assistant/Supervised/Docker/venv): Docker

## Describe the bug

While a Shelly 1 instance works without problems, the Shelly 2.5 is discovered and reports back its status like power consumption, but switching the relay doesn't work.

## Steps to Reproduce

Use the Switch in HomeAssistant, error message in HA:
Failed to call service switch/turn_on. 'switch.shelly_shsw_25_xxxxxxxx_2'

## Traceback/Error logs

```
2020-06-07 20:36:39 ERROR (MainThread) [homeassistant.components.websocket_api.http.connection.139653253548752] 'switch.shelly_shsw_25_68c63af9d4cd_2'


Traceback (most recent call last):


  File "/usr/src/homeassistant/homeassistant/components/websocket_api/commands.py", line 130, in handle_call_service


    connection.context(msg),


  File "/usr/src/homeassistant/homeassistant/core.py", line 1253, in async_call


    task.result()


  File "/usr/src/homeassistant/homeassistant/core.py", line 1288, in _execute_service


    await handler.func(service_call)


  File "/usr/src/homeassistant/homeassistant/helpers/entity_component.py", line 213, in handle_service


    self._platforms.values(), func, call, required_features


  File "/usr/src/homeassistant/homeassistant/helpers/service.py", line 419, in entity_service_call


    entity_ids.remove(entity.entity_id)


KeyError: 'switch.shelly_shsw_25_xxxxxxxxx_2'

```

**Comment by chemelli74** [comment]
Did you configure the plugin trough configuration.yaml or UI ?

Simone

**Comment by chemelli74** [comment]
Hi all, just to inform that after 3 weeks of inactivity on thread, I'll proceed to close them.
If later one someone verify that the issue is still there, a new issue need to be opened.

Thank you,

Simone

**Comment by chemelli74** [comment]
No update for 21 days, closing

Simone

**Comment by jjphoto** [comment]
Hi there, I'm having this issue - should I open a new bug, or just comment my details on this one? What's most efficient?

I can use the Web UI for a number of my Shelly 2.5s, and their status updates in HA, but I can't reliably control them in HA. They work VERY  VERY rarely. Here's my log file:

`Logger: homeassistant.components.websocket_api.http.connection.140406332127264
Source: helpers/service.py:419
Integration: Home Assistant WebSocket API (documentation, issues)
First occurred: 1:08:21 PM (3 occurrences)
Last logged: 1:09:42 PM

'switch.patio_lights_shelly'
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/components/websocket_api/commands.py", line 125, in handle_call_service
    await hass.services.async_call(
  File "/usr/src/homeassistant/homeassistant/core.py", line 1250, in async_call
    task.result()
  File "/usr/src/homeassistant/homeassistant/core.py", line 1285, in _execute_service
    await handler.func(service_call)
  File "/usr/src/homeassistant/homeassistant/helpers/entity_component.py", line 208, in handle_service
    await self.hass.helpers.service.entity_service_call(
  File "/usr/src/homeassistant/homeassistant/helpers/service.py", line 419, in entity_service_call
    entity_ids.remove(entity.entity_id)
KeyError: 'switch.patio_lights_shelly'`

**Comment by rousveiga** [comment]
I had the same problem. I had gotten the configuration instructions mixed up, and tried to configure the integration both by GUI and YAML. I rolled back the changes (deleted the new YAML and the integration), and repeated the process of installing the integration.

One thing that came to my attention is that I hadn't configured the IP of my HA hub. I added that, unchecked the checkbox about communicating with Shelly Cloud and left everything else as default. That fixed it for me.
